components/validator/validator.go

```diff
@@ -113,34 +119,46 @@ func NewValidator(parentCtx context.Context, cfg Config, l log.Logger, m metrics
 	}, nil
 }
 
-func (v *Validator) Start() {
+func (v *Validator) Start() error {
 	v.l.Info("starting Validator")
 
 	if !v.cfg.OutputSubmitterDisabled {
 		if err := v.l2os.Start(); err != nil {
-			v.l.Error("cannot start l2 output submitter: %w", err)
-			return
+			return fmt.Errorf("cannot start l2 output submitter: %w", err)
 		}
 	}
 
+	if err := v.challenger.Start(); err != nil {
+		return fmt.Errorf("cannot start challenger: %w", err)
+	}
```

Comment:
Don't we need to check the following?
```suggestion
	if !v.cfg.OutputSubmitterDisabled || !v.cfg.ChallengerDisabled {
		if err := v.challenger.Stop(); err != nil {
			return fmt.Errorf("failed to stop challenger: %w", err)
		}
	}
```
Or check it in config check function in `/components/validator/config.go` L46?

Reply:
Also in `Stop()`.